CONTRIBUTOR-DOCS/01_contributor-guides/10_typescript-configuration.md

```diff
@@ -0,0 +1,252 @@
+- **Smaller bundles**: No polyfills needed for modern JavaScript features
+- **Better debugging**: Generated code is closer to source code
+- **Already in use**: The codebase already uses ES2020 features extensively (201+ optional chaining, 158K+ nullish coalescing operations)
+
+### Browser and runtime support
+
+**Target browsers** (from `.browserslistrc`):
+
+```
+defaults
+last 2 Chrome versions
+last 2 Firefox versions
+last 2 Safari versions
+last 2 iOS versions
+not IE 11
+```
+
+**Node.js requirement**: `>=20.19.0` (defined in root `package.json`)
+
+All of these targets fully support ES2020 features.
+
+## Configuration structure
+
+### Three-tier hierarchy
+
+The TypeScript configuration uses a three-tier inheritance structure:
+
+```
+tsconfig.base.json              ← Root: Shared ES2020 settings
+├── 1st-gen/tsconfig.json       ← 1st-gen: Node resolution
+│   └── packages/*/tsconfig.json ← Package-specific overrides
+└── 2nd-gen/tsconfig.json       ← 2nd-gen: Bundler resolution
+    └── packages/*/tsconfig.json ← Package-specific overrides
+```
+
+**Benefits of this structure:**
+
+- Single source of truth for common settings
+- Clear documentation of intentional differences
+- Easy to update TypeScript version or shared settings
+- Reduced configuration duplication
+
+**Shared settings in `tsconfig.base.json`:**
+
+- `target: "ES2020"` and `lib: ["ES2020", "DOM", "DOM.Iterable"]`
+- Strict type checking rules (`strict`, `noImplicitReturns`, `noUnusedLocals`, etc.)
+- Decorator configuration (`experimentalDecorators`, `useDefineForClassFields`)
+- Common output settings (`declaration`, `sourceMap`, `skipLibCheck`)
+
+### Module resolution differences
+
+While both generations share the ES2020 target, they use different module resolution strategies based on their build tools:
+
+| Workspace | Resolution | Build Tool | Purpose |
+|-----------|------------|------------|---------|
+| 1st-gen | `node` | TypeScript compiler | Project references, tsc incremental builds |
+| 2nd-gen | `bundler` | Vite | Modern bundler optimizations, faster HMR |
+
+Both use `module: "ESNext"` for maximum interoperability.
+
+**Why different resolution strategies?**
+
+- **1st-gen** uses the classic `node` resolution because it relies on TypeScript's project references and incremental compilation features
+- **2nd-gen** uses `bundler` resolution to take advantage of Vite's optimized module resolution and Hot Module Replacement (HMR)
+
+## Cross-generation imports
+
+### How 1st-gen imports from 2nd-gen
+
+1st-gen components can import from 2nd-gen core functionality in two ways:
+
+**Via the base package (recommended):**
+
+```typescript
+import { SpectrumElement, html, css } from '@spectrum-web-components/base';
+```
+
+**Direct core import (advanced use cases):**
+
+```typescript
+import { SpectrumElement } from '@spectrum-web-components/core/shared/base/Base.js';
+```
```

Comment:
This is not accurate. The `@spectrum-web-components/base` in 1st-gen is re-exporting from `core` to maintain backwards compatibility and avoid us having to change all its imports to instead depend on 2nd-gen.